### PR TITLE
[prometheus-crds]: CRDs provided in separate Chart

### PR DESCRIPTION
**Pull request by dacamposol**
#### What this PR does / why we need it

Using `kube-prometheus-stack` from GitOps tools like Argo CD poses a problem in the way that currently the CRDs are defined to be used.

Since there are some CRDs (like `prometheuses`) which have a lot of changes between versions, the annotations exceed the maximum size, ending up in synchronise issues which can only be solved manually (for example by forcing Argo CD to `replace` that resource instead of `applying` changes to it). This behaviour can be modified at resource level, but that would require to be able to modify the annotations of the proper CRDs... Which we can not when they are under the `crd` folder.

Due to that, I'm presenting this PR, where we introduce the [CRDs as a separate Chart](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#method-2-separate-charts) and also I'm allowing to introduce some custom annotations for compatibility with GitOps tools which can rely on those for adding custom logic on how those resources need to be applied.

For example, I would desire to have the following annotation on my CRDs for ArgoCD:

```yaml
argocd.argoproj.io/sync-options: Replace=true
```

Also, this won't break compatibility of the already existing `kube-prometheus-stack`, since it contains the CRDs in the `crd` folder (according to Helm 3 specification), so we can just avoid to install those with the `skipCRDs` flag present in almost all the GitOps tools which rely on Helm, and include the CRDs by this new Chart.

#### Which issue this PR fixes

This issue has been discussed before, but it was closed without an actual solution moreover than recommending to apply the CRDs manually through `kubectl create` or `kubectl apply --force-conflicts=true --server-side`, which is not an actual solution but more a patch.

#### Special notes for your reviewer

This is just a proposal, so I acknowledge that some changes should be made before it's being accepted to be included.

#### Checklist

- [X] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [X] Title of the PR starts with chart name (new Chart)

Signed-off-by: Daniel Campos Olivares <dacamposol@gmail.com>

### CODE REVIEW

**Comment by mrueg** [comment]
I would suggest to rename the chart to prometheus-operator-crds as prometheus can run without those.

**Comment by dacamposol** [comment]
> I would suggest to rename the chart to prometheus-operator-crds as prometheus can run without those.

Fair enough 😄

**Review by monotek — CHANGES_REQUESTED** [review]
Maintainers are missing.
@desaintmartin do you want to co maintain?

**Comment by desaintmartin** [comment]
Indeed, I am interested! It would be even more awesome to have it automated somehow, discussing in https://github.com/prometheus-community/helm-charts/issues/2612.

**Comment by desaintmartin** [comment]
There are still linting errors @dacamposol.

File: charts/prometheus-operator-crds/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+apiVersion: v2
+type: application
+version: 0.1.0
+name: prometheus-operator-crds
+icon: https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png
+description: |
+  A Helm chart that collects custom resource definitions (CRDs) from the Prometheus Operator, allowing for seamless
```
Comment (jcpunk) [inline charts/prometheus-operator-crds/Chart.yaml:7]: Resolving the trailing spaces here might cleanup the linting error.

**Comment by dacamposol** [comment]
> There are still linting errors @dacamposol.

Resolved. 

Could you please trigger the tests?

**Comment by desaintmartin** [comment]
Looks good! Should we merge this? And my first contribution will be to update them. :)

**Review by desaintmartin — APPROVED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**